zipline/utils/memoize.py

```diff
@@ -1,11 +1,13 @@
 """
 Tools for memoization of function results.
 """
-from collections import OrderedDict
-from functools import wraps, partial
+from collections import OrderedDict, Sequence
+from functools import wraps
+from itertools import compress
 from weakref import WeakKeyDictionary, ref
 
 from six.moves._thread import allocate_lock as Lock
+from toolz.sandbox import unzip
```

Comment:
we also have a version in zipline.utils.functional.unzip, not sure what the difference is